src/github3/models.py

```diff
@@ -240,26 +237,28 @@ def _api(self):
             value += f"?{self._uri.query}"
         return value
 
-    @staticmethod
-    def _uri_parse(uri):
-        return requests.compat.urlparse(uri)
-
     @_api.setter
     def _api(self, uri):
         if uri:
             self._uri = self._uri_parse(uri)
         self.url = uri
 
+    @staticmethod
+    def _uri_parse(uri):
+        return requests.compat.urlparse(uri)
+
     def _iter(
         self,
         count: int,
         url: str,
-        cls: t.Type[T],
-        params: t.Optional[t.Mapping[str, t.Optional[str]]] = None,
+        cls: t.Type["GitHubCore"],
```

Comment:
Technically this is true, but it's not accurate. `_iter` returns a GitHubIterator that returns a given type, which we can tie to `cls`.

Reply:
Oh good point, let me see if I can fix that.